### PR TITLE
DOC-8913: Configuration Settings rewording

### PR DESCRIPTION
**Pull request by osfameron**
The "tables" referred to are really sections.
Slightly updated wording, normalised the sections to h3,
linked to each configuration section from the intro text.

### CODE REVIEW

File: modules/ref/pages/client-settings.adoc [hunk 1]
```diff
@@ -140,7 +146,7 @@ As an alternative to specifying the certificates to trust, you can specify a cus
 See the xref:howtos:managing-connections.adoc#ssl[Connection Management] section for more details on how to set it up properly.
 
 
-== I/O Options
+=== I/O Options
```
Comment (RichardSmedley) [inline modules/ref/pages/client-settings.adoc:150]: You're losing most of the options from the rh-nav?

Comment (osfameron) [reply modules/ref/pages/client-settings.adoc:150]: Ah! Nope, hadn't noticed that...

But some options were h2, some h3, and didn't seem to be a hierarchy, so made sense to normalise them.

Could remove the "== Configuration Options" section entirely, and promote the children to h2?

Comment (RichardSmedley) [reply modules/ref/pages/client-settings.adoc:150]: h3 will be fine once we have it as a discretionary option for rh-nav.
Meanwhile h2 including the config options was just something of a least-worst compromise.

An alternative would be to use [discrete] to remove config options from the notional hierarchy but I don't think that's any better.

there is definitely a grouping there, so removing that section feels a little odd....
although I could be persuaded (as we will end up with a compromise for now, whatever is done).

Comment (RichardSmedley) [reply modules/ref/pages/client-settings.adoc:150]: Okay, let's give it a go.

**Review by RichardSmedley — APPROVED** [review]
Trying with h3s  :+1:

**Comment by osfameron** [comment]
Fixed TOC menu:

Screenshot 2021-11-03 at 17 28 22

This requires:
* https://github.com/couchbase/docs-sdk-common/pull/36
* https://github.com/couchbase/docs-ui/pull/126

**Comment by osfameron** [comment]
Merging as docs-ui and docs-site (and docs-sdk-common) pieces are all ready to go

**Pull request merged**